app/src/androidTest/java/org/wikipedia/pageobjects/SearchPage.kt

```diff
@@ -0,0 +1,77 @@
+package org.wikipedia.pageobjects
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.NoMatchingViewException
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.hasDescendant
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withParent
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import org.hamcrest.Matchers
+import org.wikipedia.R
+
+class SearchPage : BasePage() {
+    private val searchBarMatcher = ViewMatchers.withHint("Search Wikipedia")
```

Comment:
Dalej masz trochę pomieszane importy :)
Tutaj jest `ViewMatchers.withHint()` a poniżej bezpośrednio `withId()`. To może być na później ale warto byłoby uwspólnić sposób importowania tych metod.

Dla przykładu, w projektach my przyjęliśmy kilkukrotnie taką zasadę że jeżeli coś pochodzi z bibliotek standardowych lub trzecich, to importujemy bezpośrednio (mamy więc `withId()` czy `onView()` a nie `ViewMatchers.withId()` lub `Espresso.onView()`) a gdy mamy jakieś nasze rozwiązania np. 'myCustomMatcher()` to importujemy z nazwą klasy np. `CustomMatchers.myCustomMatcher()` tak aby oddzielać trochę rzeczy napisane w frameworku od tych zaimportowanych z bilbiotek. Ale to kosmetyka :)

Najważniejsze aby być spójnym - dzięki temu kod wygląda tak jakby go pisała 1 osoba :)